### PR TITLE
Add Device Defender Integration Tests

### PR DESCRIPTION
**Pull request by shangabl**
### Motivation
- Need integration tests for Device Defender

#### Change summary
Added Integration tests for DD
Fixed typo in ubi8 docker image tag
Added a Download File integration test for Jobs

#### Revision diff summary
If there is more than one revision, please explain what has been changed since the last revision.

### Testing
 **Is your change tested? If not, please justify the reason.**  
 **Please list your testing steps and test results.** 
- CI test run result: <link>

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: integration-tests/source/devicedefender/DeviceDefenderIntegrationTests.cpp [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        Aws::ShutdownAPI(options);
+    }
+    unique_ptr<IntegrationTestResourceHandler> resourceHandler;
+    vector<std::string> metrics{"aws:all-bytes-in", "aws:all-bytes-out", "aws:all-packets-in", "aws:all-packets-out"};
+};
+
+/**
+ * To test Device Defender we are creating a Security Profile to create violations for any metrics output and using this
+ * to facilitate testing this is a simple test to verify the metrics are being emitted by the Device Client by verifying
+ * all metrics are causing violations
```
Comment (RogerZhongAWS) [inline integration-tests/source/devicedefender/DeviceDefenderIntegrationTests.cpp:49]: Just to be clear we are just trivially checking in bytes/packets in-out is less than 1 which should trigger a violation?

File: integration-tests/source/IntegrationTestResourceHandler.cpp [hunk 2]
```diff
@@ -323,3 +329,102 @@ std::string IntegrationTestResourceHandler::GetResourceId(const std::string &res
...
+{
+    string allThingsArn = targetArn.substr(0, targetArn.find_last_of(':') + 1) + "all/things";
+
+    AttachSecurityProfileRequest request;
+    request.SetSecurityProfileName(profileName);
+    request.SetSecurityProfileTargetArn(allThingsArn);
+
+    AttachSecurityProfileOutcome outcome = iotClient.AttachSecurityProfile(request);
+
+    if (!outcome.IsSuccess())
```
Comment (RogerZhongAWS) [inline integration-tests/source/IntegrationTestResourceHandler.cpp:404]: duplicate conditional

File: integration-tests/source/IntegrationTestResourceHandler.cpp [hunk 3]
```diff
@@ -323,3 +329,102 @@ std::string IntegrationTestResourceHandler::GetResourceId(const std::string &res
...
+        printf(
+            "Failed to create Security Profile: %s\n%s\n",
+            profileName.c_str(),
+            outcome.GetError().GetMessage().c_str());
+    }
+
+    AttachSecurityProfile(profileName);
+}
+
+void IntegrationTestResourceHandler::AttachSecurityProfile(const std::string &profileName)
```
Comment (RogerZhongAWS) [inline integration-tests/source/IntegrationTestResourceHandler.cpp:394]: Curious to know whether we will ever call this method on its own (not as a part of CreateAndAttachSecurityProfile). I think we either make this method private or remove the call from line 391 and call this directly from test setup.

Comment (shangabl) [reply integration-tests/source/IntegrationTestResourceHandler.cpp:394]: I think I meant to make it private but made it protected by mistake. Will change.

File: integration-tests/source/devicedefender/DeviceDefenderIntegrationTests.cpp (hunk 1 above)
Comment (shangabl) [reply integration-tests/source/devicedefender/DeviceDefenderIntegrationTests.cpp:49]: Yes I can update that to make it clearer.

This was the only way I could think to test. DD has Device Side Metrics and Cloud Side Metrics and of course we only want to test the device and don't support custom metrics. So, verifying that we are in fact emitting these metrics seemed to be the best way.

File: integration-tests/source/IntegrationTestResourceHandler.cpp [hunk 4]
```diff
@@ -202,6 +219,7 @@ void IntegrationTestResourceHandler::CleanUpThingAndCert(const std::string &thin
         DeleteCertificate(certificateId);
         DeleteThing(thingName);
     }
+    DeleteThingGroup(thingName);
```
Comment (RogerZhongAWS) [inline integration-tests/source/IntegrationTestResourceHandler.cpp:222]: is this intentional

**Review by RogerZhongAWS — DISMISSED** [review]

File: integration-tests/source/IntegrationTestResourceHandler.cpp (hunk 4 above)
Comment (shangabl) [reply integration-tests/source/IntegrationTestResourceHandler.cpp:222]: Nope... Thanks for catching this.

**Review by RogerZhongAWS — APPROVED** [review]

**Review by xlcheng1 — APPROVED** [review]

File: integration-tests/source/IntegrationTestResourceHandler.cpp [hunk 5]
```diff
@@ -323,3 +342,155 @@ std::string IntegrationTestResourceHandler::GetResourceId(const std::string &res
...
+{
+    MetricValue metricValue;
+    metricValue.SetCount(1L);
+
+    BehaviorCriteria criteria;
+    criteria.SetComparisonOperator(ComparisonOperator::less_than);
+    criteria.SetDurationSeconds(300);
+    criteria.SetValue(metricValue);
+    criteria.SetConsecutiveDatapointsToAlarm(1);
+    criteria.SetConsecutiveDatapointsToClear(1);
```
Comment (RogerZhongAWS) [inline integration-tests/source/IntegrationTestResourceHandler.cpp:384]: do you think this line can be causing any issue? would probably be safer to set this to 100 or something

Comment (shangabl) [reply integration-tests/source/IntegrationTestResourceHandler.cpp]: I hope that is the issue. Just changed! Thanks!

**Review by xlcheng1 — DISMISSED** [review]

**Review by RogerZhongAWS — APPROVED** [review]

**Review by xlcheng1 — APPROVED** [review]

**Review by RogerZhongAWS — APPROVED** [review]

**Review by xlcheng1 — APPROVED** [review]

**Pull request merged**